doc/en/user/source/styling/sld-extensions/composite-blend.rst

```diff
@@ -0,0 +1,639 @@
+.. _composite_blend:
+
+Color compositing and color blending
+=====================================
+
+Starting with GeoServer 2.7 it is possible to perform color blending and compositing, either between
+feature type styles, or by associating the blending operations to each symbolizer.
+
+GeoServer implements most of the color compositing and blending modes suggested by the `SVG compositing
+and blending level 1 specification <http://www.w3.org/TR/compositing-1/>`_.
+Either set of operations allows to control how two overlapping layers/symbols are merged togheter
+to form a final map: while normally images are just stacked on top of each other, using compositing
+and blending modes one can control how the two are going to be merged.
+
+In this section we are going to use the "source" and "destination" terms:
+
+* *source* is the image currently being painted on top of the map
+* *destination* is the current map contents, the background onto the source image is being painted to
+
+In this documentation two sources and two destination images are going to be used:
+
+Destinations (backgrounds):
+
+.. list-table::
+   :widths: 50 50
+   :header-rows: 1
+   
+   * - Destination1
+     - Destination2
+   * - .. figure:: composite-images/bkg.png
+     - .. figure:: composite-images/bkg2.png
+     
+Sources (foregrounds):
+
+.. list-table::
+   :widths: 50 50
+   :header-rows: 1
+   
+   * - Source1
+     - Source2
+   * - .. figure:: composite-images/map.png
+     - .. figure:: composite-images/map2.png
+
+
+Alpha Compositing modes
+-----------------------
+
+Alpha compositing controls how two image are merged together by using the alpha levels of the two.
+Basically, if the operation has an alpha level of 1 (the simplest case) each pixel will come from
+either the source or the destination, or none, depending on the alpha value of pixels in the source
+and destination, that is, no color mixing is being performed, but pure geometric selection instead.
+
+Here is a list of supported operations, with an explanation and visual examples. Each of the titles
+values is one key that you can use in the SLD (syntax to be described later on).
+
+copy
+....
+
+Not much of interest, only the source will be present in the output.
+
+.. list-table::
+   :widths: 50 50
+   :header-rows: 1
+   
+   * - Example1
+     - Exampe2
+   * - .. figure:: composite-images/blend1-copy.png
+     - .. figure:: composite-images/blend2-copy.png
+
+destination
+...........
+
+Not much of interest, only the destination will be present in the output
+
+.. list-table::
+   :widths: 50 50
+   :header-rows: 1
+   
+   * - Example1
+     - Exampe2
+   * - .. figure:: composite-images/blend1-destination.png
+     - .. figure:: composite-images/blend2-destination.png
+
+source-over
+...........
+
+The normal drawing, the source is painted above the destination, the destination is visible
```

Comment:
The default when drawing, the source is painted over above the destination, the destination is visible in areas where the source is transparent.